packages/shared/src/components/cards/ActionsButtons/UpvoteButtonIcon.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { ReactElement } from 'react';
+import classNames from 'classnames';
+import { feature } from '../../../lib/featureManagement';
+import { UpvoteIcon } from '../../icons';
+import { IconProps, IconSize } from '../../Icon';
+import { userPrefersReducedMotions } from '../../../styles/media';
+import { useConditionalFeature, useMedia } from '../../../hooks';
+import styles from './ActionButtons.module.css';
+
+type AnimatedButtonIconProps = IconProps & { userClicked?: boolean };
+
+const arrows = Array.from({ length: 5 }, (_, i) => i + 1);
+
+export const UpvoteButtonIcon = React.memo(function UpvoteButtonIconComp(
+  props: AnimatedButtonIconProps,
+): ReactElement {
+  const { secondary: isUpvoteActive, userClicked, ...attrs } = props;
+
+  const haveUserPrefersReducedMotions = useMedia(
+    [userPrefersReducedMotions.replace('@media', '')],
+    [false],
+    false,
+    false,
+  );
+
+  const { value: isAnimatedVersion } = useConditionalFeature({
+    feature: feature.animatedUpvote,
+    shouldEvaluate: !haveUserPrefersReducedMotions,
```

Comment:
```suggestion
    shouldEvaluate: !haveUserPrefersReducedMotions && userClicked && isUpvoteActive,
```

@ilasw let's add this as well since I confirmed it with Ido [here](https://dailydotdev.slack.com/archives/C02JAUF8HJL/p1720534108129839?thread_ts=1720532942.394519&cid=C02JAUF8HJL)